app/src/main/java/org/inspire/breath/fragments/BreathRateFragment.java

```diff
@@ -0,0 +1,111 @@
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import org.inspire.breath.R;
+import org.inspire.breath.activities.BreathRateActivity;
+import org.inspire.breath.activities.HomeActivity;
+import org.inspire.breath.data.AppRoomDatabase;
+import org.inspire.breath.data.Recording;
+import org.inspire.breath.data.RecordingDao;
+import org.inspire.breath.data.blobs.BreathTestResult;
+
+public class BreathRateFragment extends Fragment {
+
+    private BreathRateViewModel mViewModel;
+
+    public static BreathRateFragment newInstance() {
+        return new BreathRateFragment();
+    }
+    public static int count = 0;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.breath_rate_fragment, container, false);
+        TextView countdown = v.findViewById(R.id.countdowntimer);
+        TextView countAmount = v.findViewById(R.id.tapCount); //bonus points for rhyming
+        Button startRecording = v.findViewById(R.id.breathRecording);
+        Button tapTest = v.findViewById(R.id.tapTest);
+
+        startRecording.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                startRecording.setEnabled(false);
+                tapTest.setEnabled(true);
+                count = 0;
+                countAmount.setText(count + "");
+                new CountDownTimer(60000, 1000) {
+                    public void onTick(long millis) {
+                        countdown.setText((millis/1000) + "");
+                    }
+
+                    public void onFinish() {
+                        startRecording.setEnabled(true);
+                        tapTest.setEnabled(false);
+                        countdown.setText(60+"");
+                        if(count != 0) {
+                            DialogInterface.OnClickListener dialogClickListener = new DialogInterface.OnClickListener() {
+                                @Override
+                                public void onClick(DialogInterface dialog, int which) {
+                                    switch (which) { // DOUBLE BONUS POINTS FOR RHYMING
+                                        case DialogInterface.BUTTON_POSITIVE:
+                                            RecordingDao dao = AppRoomDatabase.getDatabase().recordingDao();
+                                            BreathRateViewModel viewModel = ViewModelProviders.of(getActivity()).get(BreathRateViewModel.class);
+                                            Recording r = dao.getRecordingById(viewModel.sessionID).get(0);
```

Comment:
If you use the [TestFragment](https://github.com/breathoscope/breathoscope/blob/e6fedadd2b7f9b49c5c0935ca4ca761e5d171257/app/src/main/java/org/inspire/breath/fragments/TestFragment.java) and the [TestActivity](https://github.com/breathoscope/breathoscope/blob/e6fedadd2b7f9b49c5c0935ca4ca761e5d171257/app/src/main/java/org/inspire/breath/activities/TestActivity.java) class then it will be easier for others to understand how this was done and they will all be more consistent.